x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -99,77 +97,44 @@ export async function getDocumentSources({
           rollupInterval,
           filters: [...kql, ...currentRange],
         }),
-      };
-    });
-  });
-
-  const sourcesToCheckWithSummary = [
-    ApmDocumentType.TransactionMetric as const,
-  ].flatMap((documentType) => {
-    const docTypeConfig = getConfigForDocumentType(documentType);
-
-    return (
-      enableContinuousRollups
-        ? docTypeConfig.rollupIntervals
-        : [RollupInterval.OneMinute]
-    ).flatMap((rollupInterval) => {
-      const summaryExistsFilter = {
-        bool: {
-          filter: [
-            {
-              exists: {
-                field: TRANSACTION_DURATION_SUMMARY,
-              },
-            },
-          ],
-        },
-      };
-
-      return {
-        documentType,
-        rollupInterval,
-        meta: {
-          checkSummaryFieldExists: true,
-        },
-        before: getRequest({
-          documentType,
-          rollupInterval,
-          filters: [...kql, ...beforeRange, summaryExistsFilter],
-        }),
-        current: getRequest({
+        durationSummaryCheck: getRequest({
           documentType,
           rollupInterval,
-          filters: [...kql, ...currentRange, summaryExistsFilter],
+          filters: [...kql, ...currentRange, getDurationLegacyFilter()],
         }),
       };
     });
   });
 
-  const allSourcesToCheck = [...sourcesToCheck, ...sourcesToCheckWithSummary];
-
-  const allSearches = allSourcesToCheck.flatMap(({ before, current }) => [
-    before,
-    current,
-  ]);
+  const allSearches = [
+    ...sourcesToCheck.flatMap(({ before, current, durationSummaryCheck }) => [
+      before,
+      current,
+      durationSummaryCheck,
+    ]),
+  ];
 
   const allResponses = (
     await apmEventClient.msearch('get_document_availability', ...allSearches)
   ).responses;
 
-  const checkedSources = allSourcesToCheck.map((source, index) => {
+  const checkedSources = sourcesToCheck.map((source, index) => {
     const { documentType, rollupInterval } = source;
-    const responseBefore = allResponses[index * 2];
-    const responseAfter = allResponses[index * 2 + 1];
+    const responseBefore = allResponses[index * 3];
```

Comment:
Remind me: why is this now multiplied with 3 instead of 2?

Reply:
3 queries are running for each doctype now. This constant might make things easier to read for now